packages/firebase_storage/firebase_storage/lib/src/utils.dart

```diff
@@ -31,22 +31,48 @@ Map<String, String?>? partsFromHttpUrl(String url) {
   if (decodedUrl == null) {
     return null;
   }
+  // Either a firebase storage or google cloud http url
+  String firebaseStorageHost = 'firebasestorage.googleapis.com';
+  String cloudStorageHost =
+      '(?:storage.googleapis.com|storage.cloud.google.com)';
+  String bucketDomain = r'([A-Za-z0-9.\-_]+)';
 
-  Uri uri = Uri.parse(decodedUrl);
+  if (decodedUrl.contains(firebaseStorageHost)) {
+    String version = 'v[A-Za-z0-9_]+';
+    String firebaseStoragePath = r'(/([^?#]*).*)?$';
+    RegExp firebaseStorageRegExp = RegExp(
+      '^https?://$firebaseStorageHost/$version/b/$bucketDomain/o$firebaseStoragePath',
+      caseSensitive: false,
+    );
```

Comment:
any chance to make these strings and regexp static/const?

Reply:
I've made strings `const`, cannot make `RegExp` const.

Reply:
These strings and regexp could be moved outside of the function body, so they'll be created only once, not on each function invocation